Queries - Get Started page (#377)
wormhole-foundation/wormhole-docs: products/queries/get-started.md

---
title: Get Started with Queries
description: Run your first cross-chain, verifiable query with the Wormhole Queries SDK and Proxy, using eth_call to fetch token metadata.
categories: Queries
---

# Get Started with Queries

## Introduction

[Queries](/docs/products/queries/overview) lets you fetch on-chain data from supported blockchains using `eth_call`-style requests, without submitting transactions or paying gas. The Guardian network signs the result, making it verifiable and suitable for use on-chain.

This guide walks you through requesting an API key, constructing your first query using the [Wormhole Query SDK](https://www.npmjs.com/package/@wormhole-foundation/wormhole-query-sdk){target=\_blank}, and decoding the result.

## Prerequisites

Before you begin, make sure you have the following:

 - [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm){target=\_blank} 
 - A basic understanding of JavaScript or TypeScript
 - An RPC endpoint for a supported chain (e.g., Ethereum Sepolia)
 - A Wormhole Queries API key

## Request an API Key

Wormhole Queries is in closed beta, but you can start building today.

> To interact with the system, you will use the Query Proxy, a hosted service that receives your query, routes it to the appropriate chain, and returns a signed, verifiable response from the Guardian network. This allows you to fetch on-chain data without running your own infrastructure.

Reviewer: ```suggestion
To interact with the system, you will use the Query Proxy. This hosted service receives your query, routes it to the appropriate chain, and returns a signed, verifiable response from the Guardian network. The Query Proxy allows you to fetch on-chain data without infrastructure overhead.
```

---

> title: Get Started

Reviewer 2: Please improve for SEO (see comment on NTT PR)

description: Run your first cross-chain, verifiable query with the Wormhole Queries SDK and Proxy, using eth_call to fetch token metadata.
categories: Queries
---

# Get Started

## Introduction

> Wormhole Queries let you fetch on-chain data from supported blockchains using `eth_call`-style requests, without submitting transactions or paying gas. The Guardian network signs the result, making it verifiable and suitable for use on-chain.

Reviewer 2: ```suggestion
Wormhole Queries lets you fetch on-chain data from supported blockchains using `eth_call`-style requests, without submitting transactions or paying gas. The Guardian network signs the result, making it verifiable and suitable for use on-chain.
```

Reviewer 2: First mention of Queries, we should add a link to the overview page so users can learn more

Reviewer 2: Are we calling it "Wormhole Queries" now or can we just call it "Queries"?

> This guide walks you through requesting an API key, constructing your first query using the Wormhole Query SDK, and decoding the result.

Reviewer 2: Link to the Wormhole Query SDK?

## Prerequisites

Before you begin, make sure you have the following:

 - [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm){target=\_blank} 
 - A basic understanding of JavaScript or TypeScript
 - An RPC endpoint for a supported chain (e.g., Ethereum Sepolia)
 - A Wormhole Queries API key

## Request an API Key

Wormhole Queries is in closed beta, but you can start building today.

> To access the Query Proxy, you’ll need an API key to authenticate when sending requests and receiving signed responses from the Guardian network.

Reviewer 2: What's the Query Proxy? Why do they need to or want to access it?

> To request access, join the beta by filling out the [access form](https://forms.clickup.com/45049775/f/1aytxf-10244/JKYWRUQ70AUI99F32Q){target=\_blank}. You’ll receive an API key via email once approved.

Reviewer 2: Is this manual? How long on average can this take?

Author: Yes, that is a manual process. I'm not sure, but I could ask, should I add the estimated time there?

Reviewer: I just submitted one so let's see!

## Set Up Your Project

> 1. Create a new directory for your script and initialize a `Node.js` project:

Reviewer 2: ```suggestion
1. Create a new directory for your script and initialize a Node.js project:
```

    ```ts
    --8<-- "code/products/queries/get-started/snippet-1.ts:30:42"
    ```

8. **Decode the response** - parse the signed response and extract the result of the `eth_call` to display the token name

    ```ts
    --8<-- "code/products/queries/get-started/snippet-1.ts:44:51"
    ```

???- code "Complete `query.ts`"
    ```ts
    --8<-- "code/products/queries/get-started/snippet-1.ts"
    ```

## Run Your First Query

With your script in place, run the following command from your queries directory:

```bash
API_KEY=<INSERT_QUERIES_API_KEY> npx tsx query.ts
```

???- code "Expected Output"
    ```bash
    Parsed chain response:
    EthCallQueryResponse {
    blockNumber: 8193548n,
    blockHash: '0xef97290e043a530dd2cdf2d4c513397495029cdf2ef3e916746c837dadda51a8',
    blockTime: 1745595132000000n,
    results: [
        '0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000045553444300000000000000000000000000000000000000000000000000000000'
    ]
    }

    Token name: USDC

>     ```

Reviewer 2: Terminal element?

>     ```

Reviewer 2: Terminal element?

## Next Steps

Now that you've successfully run your first verifiable query, you are ready to go deeper. Check out the following guides to build on what you've learned:

 - [Query Solana](https://github.com/wormhole-foundation/demo-queries-ts/blob/main/src/query_solana_stake_pool.ts){target=\_blank} – try fetching Solana stake pools to see how cross-chain queries apply beyond EVM
 - [Construct a Query](){target=\_blank} – understand how to build different query types using the SDK manually
 - [Verify a Query Response On-Chain](){target=\_blank} – use signed results in smart contracts

> Browse the [Supported Networks](){target=\_blank} to see where you can query today.

Reviewer 2: Let's add TODO in these links so we don't forget about them

---
title: Get Started
description: Run your first cross-chain, verifiable query with the Wormhole Queries SDK and Proxy, using eth_call to fetch token metadata.
categories: Queries
---

# Get Started

## Introduction

Wormhole Queries let you fetch on-chain data from supported blockchains using `eth_call`-style requests, without submitting transactions or paying gas. The Guardian network signs the result, making it verifiable and suitable for use on-chain.

This guide walks you through requesting an API key, constructing your first query using the Wormhole Query SDK, and decoding the result.

## Prerequisites

Before you begin, make sure you have the following:

 - [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm){target=\_blank} 
 - A basic understanding of JavaScript or TypeScript
 - An RPC endpoint for a supported chain (e.g., Ethereum Sepolia)

>  - A Wormhole [Queries API key](#request-an-api-key)

Reviewer: I don't think making this a link is needed since the item is directly beneath it
